### PR TITLE
[Docs] Convert to markdown cond.rst, config_mod.rst

### PR DESCRIPTION
**Pull request by windsonsea**
FIxes #155014 

Only included 2 files in this PR:

- cond.rst
- config_mod.rst

cc @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

**Comment by windsonsea** [comment]
@pytorchbot label "topic: not user facing"

This is my local bash history. Hope the git log remained.
```
 1426  git checkout main
 1427  git pull -r
 1429  git checkout -b bttcond
 1430  cd docs/source

 1431  git mv bottleneck.rst bottleneck.md
 1432  git mv checkpoint.rst checkpoint.md
 1433  git mv complex_numbers.rst complex_numbers.md
 1434  git mv cond.rst cond.md
 1435  git mv config_mod.rst config_mod.md

 1436  git status
 1437  git add .
 1438  git commit -m "[Docs] Convert to markdown to fix #155014"
 1439  git push origin bttcond
```

File: docs/source/cond.md [hunk 1]
```diff
@@ -1,41 +1,39 @@
-.. _cond:
+(_cond)=
```
Comment (svekars) [inline docs/source/cond.md]:
```suggestion
(cond)=
```

File: docs/source/complex_numbers.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
+(_complex_numbers-doc)=
```
Comment (svekars) [inline docs/source/complex_numbers.md]:
```suggestion
(complex_numbers-doc)=
```

File: docs/source/cond.md [hunk 3]
```diff
@@ -62,29 +60,29 @@ Below is an example that uses cond to branch based on input shape:
...
     dim_batch = torch.export.Dim("batch", min=2)
     ep = torch.export.export(DynamicShapeCondPredicate(), (inp,), {}, dynamic_shapes={"x": {0: dim_batch}})
     print(ep)
+```
 
 This gives us an exported program as shown below:
 
-.. code-block::
-
+```python
```
Comment (svekars) [inline docs/source/cond.md]: Should not be [python](https://github.com/pytorch/pytorch/actions/runs/15574976171/job/43860492003?pr=155653#step:11:977).

File: docs/source/cond.md [hunk 4]
```diff
@@ -122,11 +120,11 @@ Here is another example that showcases how to express a data-dependent control f
 
         def forward(self, x: torch.Tensor) -> torch.Tensor:
             return torch.cond(x.sum() > 4.0, true_fn, false_fn, (x,))
+```
 
 The exported program we get after export:
 
-.. code-block::
-
+```python
```
Comment (svekars) [inline docs/source/cond.md]: Should not be [python](https://github.com/pytorch/pytorch/actions/runs/15574976171/job/43860492003?pr=155653#step:11:977)

**Review by svekars — COMMENTED** [review]
Looking good. Added comments that should help to fix the build.

**Comment by svekars** [comment]
Files changed Created on dates:
* https://docs-preview.pytorch.org/pytorch/pytorch/155653/complex_numbers.html
* https://docs-preview.pytorch.org/pytorch/pytorch/155653/checkpoint.html
* https://docs-preview.pytorch.org/pytorch/pytorch/155653/bottleneck.html

You can try to fix like this or start from scratch on a new branch:
1. Save your changes to .md somewhere.
2. On this branch undo your changes to all the affected files and delete .md files. From docs dir: git checkout origin/main -- source/complex_numbers.rst  - and other files. 
3. Then git mv source/complex_numbers.rst source/complex_numbers.md and commit. Do for each file.
5. Can check git log: git log -- follow -- source/complex_numbers.md - this should show you a bunch of commits.
6. Push into this PR the above commits.
7. Copy paste changes you've saved in step 1 into your .md files. Commit/push into this PR.

**Comment by windsonsea** [comment]
Hi @svekars, understood.

Will remove these 3 files from this PR and open a new branch and PR to add them.

> `git mv` may have some bugs that cause the git-log lost -_!

**Review by svekars — APPROVED** [review]

**Comment by svekars** [comment]
Need to fix code blocks indentation in a follow up PR.
